src/sdk/cohort.ts

```diff
@@ -0,0 +1,79 @@
+import { PublicKey } from '@nucypher/nucypher-core';
+
+import { Porter } from '../characters/porter';
+import { ChecksumAddress } from '../types';
+
+export type Ursula = {
+  readonly checksumAddress: ChecksumAddress;
+  readonly uri: string;
+  readonly encryptingKey: PublicKey;
+};
+interface CohortParameters {
+  porterUri: string;
+  threshold: number;
+  shares?: number;
+  include?: ChecksumAddress[];
+  exclude?: ChecksumAddress[];
+}
+
+interface CohortJSON {
+  ursulaAddresses: ChecksumAddress[];
+  threshold: number;
+  porterUri: string;
+}
+
+export class Cohort {
+  constructor(
+    public readonly ursulaAddresses: ChecksumAddress[],
+    public readonly threshold: number,
+    public readonly porterUri: string
+  ) {
+    this.ursulaAddresses = ursulaAddresses;
+    this.threshold = threshold;
+    this.porterUri = porterUri;
+  }
+
+  public static async create({
+    porterUri,
+    threshold,
+    shares = 0,
+    include = [],
+    exclude = [],
+  }: CohortParameters) {
+    if (shares == 0 && include.length == 0) {
+      throw new TypeError('Shares is 0 and Include is an empty array');
+    }
+    if (shares == 0 && include.length > 0) {
+      shares = include.length;
+    }
```

Comment:
Ah, interesting.

The way `include` works in `nucypher-ts` is that it forces Porter to always contact certain Ursulas, but not exclusively. Shall we reuse this protocol, or does it make sense to change how it works?

Reply:
i think that's the expected behaviour. If i set include to be the set of 5 nodes that I want to use (I don't want more than that) - then Porter will only contact them and they'll be the cohort right?

Reply:
So the cohort defines Ursulas _exclusively_ - I think that will make sense for what `Cohort` is supposed to represent,

Reply:
What happens if `shares > ursulas.length` thought? Where would the "extra shares" go?

Reply:
> The way include works in nucypher-ts is that it forces Porter to always contact certain Ursulas, but not exclusively. 

This is sensible, not need to change it right now.  There is room for refinement, for example, what if the *include* nodes cannot be contacted (and therefor cannot be included) but these are best suited for an issue at the current development stage.

Reply:
If it's a similar amount of work to adjust Porter to fit the static node list design, as it is to adjust the SDK's Cohort definition as per my other comments, then we should choose the longer-term option – the superior design imo being resampling at data share event and separating out `include` so it's not part of global configs. This is a fairly important decision so deserves a voice chat/debate :)

Reply:
If it's a similar amount of work to adjust Porter to fit the static node list design, as it is to adjust the SDK's Cohort definition as per my other comments, then we should choose the longer-term option.